### PR TITLE
document IPSetBuilder API more, make IPSetBuilder.IPSet clear errors

### PR DESCRIPTION
**Pull request by josharian**

### CODE REVIEW

**Review by mdlayher — APPROVED** [review]
LGTM.

File: ipset.go [hunk 1]
```diff
@@ -287,6 +299,7 @@ func (s *IPSetBuilder) IPSet() (*IPSet, error) {
 	if len(s.errs) == 0 {
 		return ret, nil
 	} else {
+		s.errs = s.errs[:0:0]
```
Comment (bradfitz) [inline ipset.go:302]: This seems wrong. You're clearing it before returning it?

Comment (josharian) [reply ipset.go]: Oops. Fixed, thanks. The test that would have caught this bug is in #188.

**Comment by josharian** [comment]
I'm going AFK in a few hours. Please feel free to edit and merge at will.

**Pull request merged**